pkg/config/expand_test.go

```diff
@@ -447,64 +448,88 @@ func (s *MySuite) TestIdentifySimpleVariable(c *C) {
 
 func (s *MySuite) TestExpandSimpleVariable(c *C) {
 	// Setup
-	testModID := "existingModule"
-	testModule := Module{
-		ID:     testModID,
+	testModule0 := Module{
+		ID:     "module0",
+		Kind:   "terraform",
+		Source: "./module/testpath",
+	}
+	testModule1 := Module{
+		ID:     "module1",
 		Kind:   "terraform",
 		Source: "./module/testpath",
 	}
 	testBlueprint := Blueprint{
-		BlueprintName: "",
+		BlueprintName: "test-blueprint",
 		Vars:          make(map[string]interface{}),
-		DeploymentGroups: []DeploymentGroup{{
-			Name:             "",
-			TerraformBackend: TerraformBackend{},
-			Modules:          []Module{testModule},
-		}},
+		DeploymentGroups: []DeploymentGroup{
+			{
+				Name:             "zero",
+				TerraformBackend: TerraformBackend{},
+				Modules:          []Module{testModule0},
+			},
+			{
+				Name:             "one",
+				TerraformBackend: TerraformBackend{},
+				Modules:          []Module{testModule1},
+			},
+		},
 		TerraformBackendDefaults: TerraformBackend{},
 	}
-	testVarContext := varContext{
+
+	testVarContext0 := varContext{
 		blueprint:  testBlueprint,
 		modIndex:   0,
 		groupIndex: 0,
 	}
-	testModToGrp := make(map[string]int)
+
+	testVarContext1 := varContext{
+		blueprint:  testBlueprint,
+		modIndex:   0,
+		groupIndex: 1,
+	}
+
+	// strictly speaking this unit test now also tests this function
+	// we should expand the blueprint used in config_test.go to include
+	// multiple deployment groups
+	testModToGrp, err := checkModuleAndGroupNames(testBlueprint.DeploymentGroups)
+	c.Assert(err, IsNil)
+	c.Assert(testModToGrp[testModule0.ID], Equals, 0)
+	c.Assert(testModToGrp[testModule1.ID], Equals, 1)
 
 	// Invalid variable -> no .
-	testVarContext.varString = "$(varsStringWithNoDot)"
-	_, err := expandSimpleVariable(testVarContext, testModToGrp)
+	testVarContext1.varString = "$(varsStringWithNoDot)"
+	_, err = expandSimpleVariable(testVarContext1, testModToGrp)
 	expectedErr := fmt.Sprintf("%s.*", errorMessages["invalidVar"])
 	c.Assert(err, ErrorMatches, expectedErr)
 
 	// Global variable: Invalid -> not found
-	testVarContext.varString = "$(vars.doesntExists)"
-	_, err = expandSimpleVariable(testVarContext, testModToGrp)
+	testVarContext1.varString = "$(vars.doesntExists)"
+	_, err = expandSimpleVariable(testVarContext1, testModToGrp)
 	expectedErr = fmt.Sprintf("%s: .*", errorMessages["varNotFound"])
 	c.Assert(err, ErrorMatches, expectedErr)
 
 	// Global variable: Success
-	testVarContext.blueprint.Vars["globalExists"] = "existsValue"
-	testVarContext.varString = "$(vars.globalExists)"
-	got, err := expandSimpleVariable(testVarContext, testModToGrp)
+	testVarContext1.blueprint.Vars["globalExists"] = "existsValue"
+	testVarContext1.varString = "$(vars.globalExists)"
+	got, err := expandSimpleVariable(testVarContext1, testModToGrp)
 	c.Assert(err, IsNil)
-	expected := "((var.globalExists))"
-	c.Assert(got, Equals, expected)
+	expectedErr = "((var.globalExists))"
```

Comment:
Why `expectedErr` if this isn't an error?

Reply:
Resolved on HEAD